### PR TITLE
accumulator: Add/clean-up comments. Privatize some variables/functions

### PR DESCRIPTION
**Pull request by kcalvinalvin**
This commit mostly makes the package accumulator well maintained.
Stale comments were either removed or improved. Some variables and
functions that have no use to someone that would be looking to import
this package was privatized. This helps with readability on pkg.go.dev
(aka godoc.org).

### CODE REVIEW

File: accumulator/batchproof.go [hunk 1]
```diff
@@ -7,24 +7,37 @@ import (
...
 
-/*
-Batchproof serialization is:
-4bytes numTargets
-4bytes numHashes
-[]Targets (8 bytes each)
-[]Hashes (32 bytes each)
-*/
+	// All the nodes in the tree that are needed to hash up to the root of
+	// the tree. If Targets are [00, 01], then Proof would be [00, 01, 04, 05, 06].
```
Comment (dergoegge) [inline accumulator/batchproof.go:23]: The proof for [00, 01] would actually just be [00, 01, 05] the other values you can compute from those.

Comment (dergoegge) [reply accumulator/batchproof.go:23]: Ideally you wouldnt even need the targets in there so just [05].

File: accumulator/pollard.go [hunk 2]
```diff
@@ -14,33 +14,45 @@ var ErrorStrings = map[uint32]error{
...
+	numLeaves uint64
+
+	// roots are the slice of the tree roots ordered in big to small
+	// (right to left of the forest).
+	//
+	// NOTE: Since roots don't have nieces, they point to children.
+	// In the below tree, 06 is the root and it points to its children,
+	// 04 and 05. However, 04 points to 02 and 03; 05 points to 00 and 01.
+	// They point to their nieces. The leaves are also different in that
+	// they don't point to anything.
```
Comment (dergoegge) [inline accumulator/pollard.go:30]: Could mention here that the leaves point to something if they are cached

File: accumulator/readme.md [hunk 3]
```diff
@@ -1,12 +1,65 @@
...
+Package accumulator provides a general purpose dynamic accumulator. There are two main structs: Forest and Pollard.
 
-## flow
+The Forest contains the entire utreexo accumulator (all the nodes in the forest), and can be used to produce inclusion-proofs for Pollards to verify. Pollard contains a partially populated accumulator and can verify inclusion-proofs from the Forest. A Pollard *can* contain the entire accumulator. A Forest *must* contain everything.
 
-Forest can be initialized and then have Modify() called repeatedly with hashes to add and *positions* to delete.  If you need to remember the positions of the things to delete based on their hashes, ProveBlock() will provide that data.
+Installation
```
Comment (dergoegge) [inline accumulator/readme.md:10]: This section seems incomplete

File: accumulator/batchproof.go (hunk 1 above)
Comment (kcalvinalvin) [reply accumulator/batchproof.go:23]: Ah ok thanks. Will fix

File: accumulator/batchproof.go [hunk 4]
```diff
@@ -7,24 +7,38 @@ import (
...
-/*
-Batchproof serialization is:
-4bytes numTargets
-4bytes numHashes
-[]Targets (8 bytes each)
-[]Hashes (32 bytes each)
-*/
+	// All the nodes in the tree that are needed to hash up to the root of
+	// the tree. If Targets are [00, 01], then Proof would be [00, 01, 05].
+	// TODO: Remove targets from the proof as it is redundant.
```
[the remark this answers is not in the document]
Comment (kcalvinalvin) [reply accumulator/batchproof.go:24]:
```
func TestHiHi(t *testing.T) {
        // Create forest in memory
        f := NewForest(nil, false, "", 0)

        // last index to be deleted. Same as blockDels
        lastIdx := uint64(7)

        // Generate adds
        adds := make([]Leaf, 8)
        adds[0].Hash = Hash{1}
        adds[1].Hash = Hash{2}
        adds[2].Hash = Hash{3}
        adds[3].Hash = Hash{4}
        adds[4].Hash = Hash{5}
        adds[5].Hash = Hash{6}
        adds[6].Hash = Hash{7}
        adds[7].Hash = Hash{8}

        // Modify with the additions to simulate txos being added
        _, err := f.Modify(adds, nil)
        if err != nil {
                t.Fatal(err)
        }

        // create blockProof based on the last add in the slice
        blockProof, err := f.ProveBatch(
                []Hash{adds[lastIdx].Hash})

        if err != nil {
                t.Fatal(err)
        }

        fmt.Println("blockProof.Proof", blockProof.Proof)
        fmt.Println("blockProof.Targets", blockProof.Targets)

        // Confirm that verify block proof works
        shouldBetrue := f.VerifyBatchProof(blockProof)
        if shouldBetrue != true {
                t.Fail()
                t.Logf("Block failed to verify")
        }

        // delete last leaf and add a new leaf
        adds = make([]Leaf, 1)
        adds[0].Hash = Hash{9}
        _, err = f.Modify(adds, []uint64{lastIdx})
        if err != nil {
                t.Fatal(err)
        }

        // Attempt to verify block proof with deleted element
        shouldBeFalse := f.VerifyBatchProof(blockProof)
        if shouldBeFalse != false {
                t.Fail()
                t.Logf("Block verified with old proof. Double spending allowed.")
        }
}
```

This code outputs 
```
blockProof.Proof [[7 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [8 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [156 219 14 183 67 32 18 158 168 219 140 124 32 188 40 13 128 78 210 129 116 112 216 38 194 67 116 55 171 238 191 138] [130 101 80 136 216 253 117 103 105 208 152 138 108 236 5 231 72 239 20 203 114 186 183 16 62 146 14 83 49 165 199 168]]
blockProof.Targets [7]
```

The target seems to still be in the proofs.

[the remark this answers is not in the document]
Comment (adiabat) [reply accumulator/batchproof.go:24]: Aha - that's been finished in the hashtoknown branch but not merged in here.  So yeah it's still "todo" on this branch

**Pull request merged**